AMO and DevHub homepage shared footer

AlexandraMoga opened the pull request:
I started working on adding tests for the Developer Hub - https://addons.allizom.org/developers/ - homepage as part of my Q1 project goal and I've noticed that its main footer  has  mostly the same locators as the AMO footer (except for the sections crossed out in the screenshot).

![image](https://user-images.githubusercontent.com/31961530/110954995-c8d62880-8351-11eb-8d66-989576091e9f.png)


Since we already have tests covering the main AMO footer in https://github.com/mozilla/addons-release-tests/blob/master/tests/test_home.py, I think we should try to use those tests for both pages, rather than adding new almost identical tests for DevHub.

I'm not sure if my approach is correct or if there is a better way to implement this, hence I'm making this a draft PR.

AlexandraMoga commented on tests/test_home.py:
```diff
@@ -105,12 +109,14 @@ def test_mozilla_footer_link(base_url, selenium):
 
+@pytest.mark.parametrize('page', ['', '/developers'])
 @pytest.mark.parametrize(
     "count, links",
```
I'm using stack parameterization here because I couldn't make this work from only one `@pytest.mark.parametrize` decorator.

jrbenny35 replied:
I think this is fine, pytest will figure it out appropriately.

jrbenny35 commented on tests/test_home.py:
```diff
@@ -105,12 +109,14 @@ def test_mozilla_footer_link(base_url, selenium):
-def test_addons_footer_links(base_url, selenium, count, links):
-    page = Home(selenium, base_url).open()
+def test_addons_footer_links(base_url, selenium, page, count, links):
+    selenium.get(f'{base_url}{page}')
```
New Strings!!!!

hackebrot commented on tests/test_home.py:
```diff
@@ -105,12 +109,14 @@ def test_mozilla_footer_link(base_url, selenium):
 
+@pytest.mark.parametrize('page', ['', '/developers'])
 @pytest.mark.parametrize(
     "count, links",
```
Hi there! 👋🏻 

The parameter name is plural `links` however the value will be a single string item. I suggest renaming the ID as follows to make this clearer:

```suggestion
    "count, link",
```

hackebrot commented on tests/test_home.py:
```diff
@@ -105,12 +109,14 @@ def test_mozilla_footer_link(base_url, selenium):
     page.wait_for_current_url(links)
 
 
+@pytest.mark.parametrize('page', ['', '/developers'])
```
I find explicit IDs for test parameters very useful for producing more comprehensible test reports. I blogged about this a few years ago. The described APIs should still work. Hope this helps! 📝 

https://raphael.codes/blog/ids-for-pytest-fixtures-and-parametrize/

hackebrot reviewed (commented):
Hi there! 👋🏻 

I added two suggestions. Hope you don't mind.

AlexandraMoga commented on pages/desktop/base.py:
```diff
@@ -242,7 +242,7 @@ class Footer(Region):
     _footer_amo_links_locator = (By.CSS_SELECTOR, '.Footer-amo-links')
     _footer_browsers_links_locator = (By.CSS_SELECTOR, '.Footer-browsers-links')
-    _footer_products_links_locator = (By.CSS_SELECTOR, '.Footer-product-links')
+    _footer_products_links_locator = (By.CSS_SELECTOR, '.Footer-wrapper section:nth-child(4)')
```
I found a locator that works both for AMO and DevHub. How I can check the Product footer links from the same tests for both pages.

AlexandraMoga commented:
I've addressed the review suggestions and also changed a locator to allow the Product footer links to be checked from the same test. With regards to explicit IDs for test parameters (very good suggestion by the way), I've only added them to the first parameter set (the one defining the `page`) for the time being. This has improved the readability  of the test report by making it more clear on which page footer the tests is being run.

AlexandraMoga commented:
We can't merge this PR until https://github.com/mozilla/addons-server/issues/16753 gets fixed, otherwise some DevHub footer links will be different from those on AMO.

AlexandraMoga commented:
We can close this since I went for a slightly different solution in #368
